Source/Error.swift

```diff
@@ -24,41 +24,540 @@
+        return false
+    }
+
+    var isBodyPartFilenameInvalid: Bool {
+        if case .bodyPartFilenameInvalid = self { return true }
+        return false
+    }
+
+    var isBodyPartFileNotReachable: Bool {
+        if case .bodyPartFileNotReachable = self { return true }
+        return false
+    }
+
+    var isBodyPartFileNotReachableWithError: Bool {
+        if case .bodyPartFileNotReachableWithError = self { return true }
+        return false
+    }
+
+    var isBodyPartFileIsDirectory: Bool {
+        if case .bodyPartFileIsDirectory = self { return true }
+        return false
+    }
+
+    var isBodyPartFileSizeNotAvailable: Bool {
+        if case .bodyPartFileSizeNotAvailable = self { return true }
+        return false
+    }
+
+    var isBodyPartFileSizeQueryFailedWithError: Bool {
+        if case .bodyPartFileSizeQueryFailedWithError = self { return true }
+        return false
+    }
+
+    var isBodyPartInputStreamCreationFailed: Bool {
+        if case .bodyPartInputStreamCreationFailed = self { return true }
+        return false
+    }
+
+    var isOutputStreamCreationFailed: Bool {
+        if case .outputStreamCreationFailed = self { return true }
+        return false
+    }
+
+    var isOutputStreamFileAlreadyExists: Bool {
+        if case .outputStreamFileAlreadyExists = self { return true }
+        return false
+    }
+
+    var isOutputStreamURLInvalid: Bool {
+        if case .outputStreamURLInvalid = self { return true }
+        return false
+    }
+
+    var isOutputStreamWriteFailed: Bool {
+        if case .outputStreamWriteFailed = self { return true }
+        return false
+    }
+
+    var isInputStreamReadFailed: Bool {
+        if case .inputStreamReadFailed = self { return true }
+        return false
+    }
+
+}
+
+extension AFError.SerializationFailureReason {
+
+    var isInputDataNil: Bool {
+        if case .inputDataNil = self { return true }
+        return false
+    }
+
+    var isInputDataNilOrZeroLength: Bool {
+        if case .inputDataNilOrZeroLength = self { return true }
+        return false
+    }
+
+    var isStringSerializationFailed: Bool {
+        if case .stringSerializationFailed = self { return true }
+        return false
+    }
+
+    var isJSONSerializationFailed: Bool {
+        if case .jsonSerializationFailed = self { return true }
+        return false
+    }
+
+    var isPropertyListSerializationFailed: Bool {
+        if case .propertyListSerializationFailed = self { return true }
+        return false
+    }
+
+}
+
+extension AFError.ValidationFailureReason {
+
+    var isMissingContentType: Bool {
+        if case .missingContentType = self { return true }
+        return false
+    }
+
+    var isUnacceptableContentType: Bool {
+        if case .unacceptableContentType = self { return true }
+        return false
+    }
+
+    var isUnacceptableStatusCode: Bool {
+        if case .unacceptableStatusCode = self { return true }
+        return false
+    }
+
+}
```

Comment:
I think we should move all these Boolean checks into an extension that is only part of the test target. I don't see us exposing these as public any time soon. 

Why I think it would be wise to move these out of the `Error.swift` file is that they greatly convolute the jumpbar and the file itself. There are so many APIs available that it's difficult to tell what I can and can't use right now. If we pull half the APIs out into the tests, it will greatly simplify the readability of this file.

We could move all these into an `Error+AlamofireTests.swift` file. Then we could always move them back if necessary.